File: sdkwork-backend-sdk-typescript/src/types/social-info-list.ts
import type { FacebookInfo } from './facebook-info';
import type { InstagramInfo } from './instagram-info';
import type { LinkedInInfo } from './linked-in-info';
import type { TelegramInfo } from './telegram-info';
import type { TikTokInfo } from './tik-tok-info';
import type { TwitterInfo } from './twitter-info';
import type { WechatInfo } from './wechat-info';
import type { WhatsAppInfo } from './whats-app-info';

export interface SocialInfoList {
  wechat?: WechatInfo;
  facebook?: FacebookInfo;
  twitter?: TwitterInfo;
  instagram?: InstagramInfo;
  linkedin?: LinkedInInfo;
  telegram?: TelegramInfo;
  whatsapp?: WhatsAppInfo;
  tiktok?: TikTokInfo;
}

File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/SocialInfoList.kt
package com.sdkwork.backend

data class SocialInfoList(
    val wechat: WechatInfo? = null,
    val facebook: FacebookInfo? = null,
    val twitter: TwitterInfo? = null,
    val instagram: InstagramInfo? = null,
    val linkedin: LinkedInInfo? = null,
    val telegram: TelegramInfo? = null,
    val whatsapp: WhatsAppInfo? = null,
    val tiktok: TikTokInfo? = null
)
